src/mbgl/renderer/layers/render_custom_drawable_layer.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <mbgl/style/layers/custom_drawable_layer_impl.hpp>
+#include <mbgl/renderer/render_layer.hpp>
+
+namespace mbgl {
+
+class RenderCustomDrawableLayer final : public RenderLayer {
+public:
+    explicit RenderCustomDrawableLayer(Immutable<style::CustomDrawableLayer::Impl>);
+    ~RenderCustomDrawableLayer() override;
+
+#if MLN_DRAWABLE_RENDERER
+    /// Generate any changes needed by the layer
+    void update(gfx::ShaderRegistry&,
+                gfx::Context&,
+                const TransformState&,
+                const std::shared_ptr<UpdateParameters>&,
+                const RenderTree&,
+                UniqueChangeRequestVec&) override;
+#endif
+
+private:
+    void transition(const TransitionParameters&) override {}
+    void evaluate(const PropertyEvaluationParameters&) override;
+    bool hasTransition() const override;
+    bool hasCrossfade() const override;
+    void markContextDestroyed() override;
+    void prepare(const LayerPrepareParameters&) override;
+
+#if MLN_LEGACY_RENDERER
```

Comment:
Is this layer used or included when compiling the legacy renderer?

Reply:
no, it should not. there's no dependency to it from _legacy_. I'm rechecking build files.